psc-ide/DESIGN.org

```diff
@@ -165,9 +165,11 @@
     the PROTOCOL.md file.
     
 *** Filters
-    Filters are functions of type =Map ModuleName [IdeDeclaration] -> Map
-    ModuleName [IdeDeclaration]=. We keep the =Map= structure around to make the
-    common case of filtering by module names fast.
+    Filters are functions of type =Map ModuleName [IdeDeclaration] ->
```

Comment:
Perhaps this is a bit pedantic of me, but I think it's probably safer to say that filters can be thought of as functions of that type, rather than actually being functions of that type. I think there are lots of functions of that type which aren't expressible as filters, or which wouldn't be commutative.